lasagne/tests/layers/test_conv.py

```diff
@@ -287,7 +284,7 @@ def test_init_none_nonlinearity_bias(self, Conv2DImpl, DummyInputLayer):
         assert layer.b is None
 
     def test_invalid_pad(self, Conv2DImpl, DummyInputLayer):
-        input_layer = DummyInputLayer((1, 2, 3))
+        input_layer = DummyInputLayer((1, 2, 3, 3))
```

Comment:
It's magic! Or maybe it's just because none of the convolutional layers actually bothers to check its input shape in the constructor, so it gets to the point where it runs `self.pad = as_tuple(pad, 2, int)`, which raises the `TypeError` tested for.

Checking the input dimensionality is one of the things I'd like to add after refactoring the convolutional layers to be derived from a common base class.